cognitive_architecture/database/relationaldb/database.py

```diff
@@ -0,0 +1,92 @@
+"""Database configuration and connection."""
+from pathlib import Path
+from contextlib import asynccontextmanager
+from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession
+from sqlalchemy.orm import declarative_base, sessionmaker
+from dotenv import load_dotenv
+from cognitive_architecture.config import Config
+
+
+load_dotenv()
+
+
+
+# in seconds
+MAX_RETRIES = 3
+RETRY_DELAY = 5
+
+class DatabaseConfig:
+    """Configuration for the database connection."""
+    def __init__(
+        self,
+        db_type=None,
+        db_name=None,
+        host=None,
+        user=None,
+        password=None,
+        port=None,
+    ):
+        self.config = Config()
+        self.config.load()
+        self.base_path = Path(self.config.db_path)
+        # Load default values from environment variables or use provided values
+        self.db_type = db_type or self.config.db_type
+        self.db_name = db_name or self.config.db_name
+        self.host = host or self.config.db_host
+        self.user = user or self.config.db_user
+        self.password = password or self.config.db_password
+        self.port = port or self.config.db_port
+
+
+
+    def get_sqlalchemy_database_url(self):
```

Comment:
It's always better to get data through dependency injection than to have it read from internal config or class instance. I would not use `self.config` here, but always arguments passed to the function.
Level of abstraction can be added on top of it, where we can either read it from the file or some internal config, but this function would always get it from arguments.
OOP just makes everything ugly and unnecessary complex.